### PR TITLE
Leader election migration 1: ConfigMap & Lease

### PR DESCRIPTION
**Pull request by 2uasimojo**
We were using ConfigMap-based leader election. We need to migrate to using Lease instead.

Part 1: Use ConfigMap *and* Lease so we can upgrade smoothly from a build using just ConfigMap.
Part 2: Remove ConfigMap.

This is part 1.

[HIVE-1736](https://issues.redhat.com/browse/HIVE-1736)

### CODE REVIEW

File: cmd/util/leaderelection.go [hunk 1]
```diff
@@ -25,19 +24,26 @@ func RunWithLeaderElection(ctx context.Context, cfg *rest.Config, lockNS, lockNa
...
+		lockNS,
+		lockName,
+		kubeClient.CoreV1(),
+		kubeClient.CoordinationV1(),
+		resourcelock.ResourceLockConfig{
 			Identity: id,
 		},
+	)
+	if err != nil {
+		log.Fatal("failed to create lock for leader election config")
```
Comment (2uasimojo) [inline cmd/util/leaderelection.go:46]: `.WithError(err)`

File: cmd/operator/main.go [hunk 2]
```diff
@@ -8,18 +8,13 @@ import (
...
 
-	"github.com/google/uuid"
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 
 	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
-	"k8s.io/client-go/kubernetes"
```
Comment (joelddiaz) [inline cmd/operator/main.go:19]: If we're going to clean up this package, can we also remove those two `_` imports that appear to serve no purpose? (FWIW, they're also in the manager package too...)

File: cmd/manager/main.go [hunk 3]
```diff
@@ -64,11 +60,8 @@ import (
 )
 
 const (
-	defaultLogLevel             = "info"
-	leaderElectionConfigMap     = "hive-controllers-leader"
-	leaderElectionLeaseDuration = "120s"
-	leaderElectionRenewDeadline = "90s"
-	leaderElectionRetryPeriod   = "30s"
+	defaultLogLevel         = "info"
+	leaderElectionConfigMap = "hive-controllers-leader"
```
Comment (joelddiaz) [inline cmd/manager/main.go]: Should this be 'leaderElectionLockName' (to follow the other changes in this PR)?

Comment (2uasimojo) [reply cmd/manager/main.go]: That's done in the third commit. (You must be looking at the second.)

File: cmd/operator/main.go (hunk 2 above)
Comment (2uasimojo) [reply cmd/operator/main.go:19]: Taking out magic `_` imports scares me, as they _sometimes_ seem to have implications for code generation and other not-runtime ballyhoo. Can I do that in a separate PR?

File: cmd/manager/main.go (hunk 3 above)
Comment (joelddiaz) [reply cmd/manager/main.go]: Yep. Missed it as I hoped around the commits.

**Pull request merged**